### PR TITLE
Save memory on scoreboard teams

### PR DESCRIPTION
**Pull request by mikroskeem**
By using enums of course

### CODE REVIEW

**Review by electronicboy — APPROVED** [review]

File: BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
++    // Waterfall start - save memory on scoreboard teams
++    @Deprecated
++    public String getNameTagVisibility() {
++        return nameTagVisibility.getValue();
++    }
++
++    @Deprecated
++    public void setNameTagVisibility(String value) {
++        NametagVisibility enumValue;
++        if (value == null) {
```
Comment (Janmm14) [inline BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch:37]: I don't see a reason to add any safeties with an alternate, possibly wrong state here. Suggesting to leave out the null check and throw an exception if VALUES.get() returns null (same for collisionrule).

File: BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
++        ALWAYS("always"),
++        PUSH_OTHER_TEAMS("pushOtherTeams"),
++        PUSH_OWN_TEAM("pushOwnTeam"),
++        NEVER("never"),
++        ;
++
++        @lombok.Getter
++        private final String value;
++
++        public static final java.util.Map<String, CollisionRule> VALUES = new java.util.HashMap<>();
```
Comment (Janmm14) [inline BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch]: Suggesting to use ImmutableMap or sth. similar (same for collisionrule)

File: BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch (hunk 1 above)
Comment (mikroskeem) [reply BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch:37]: Right, will do so.

Comment (Janmm14) [reply BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch:37]: I edited my comment.

File: BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch (hunk 2 above)
Comment (aikar) [reply BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch]: Eh i don't think the complexity in the change is worth that. immutable isnt gonna offer any real performance benefit.

Comment (Janmm14) [reply BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch]: Yeah, if there are 10 fields more or less in that array it won't make any difference in performance, I just thought it'd look better.

**Comment by mikroskeem** [comment]
Made requested changes.

**Pull request closed without merging**